### PR TITLE
migration process to unified syntax for various contracts

### PR DESCRIPTION
**Pull request by mihaicalinluca**

### CODE REVIEW

File: contracts/feature-tests/composability/esdt-contract-pair/first-contract/src/lib.rs [hunk 1]
```diff
@@ -90,14 +90,13 @@ pub trait FirstContract {
...
-            &esdt_value,
-            self.blockchain().get_gas_left(),
-            &ManagedBuffer::from(SECOND_CONTRACT_REJECT_ESDT_PAYMENT),
-            &ManagedArgBuffer::new(),
-        );
+        let gas_left = self.blockchain().get_gas_left();
+        self.tx()
+            .to(&second_contract_address)
+            .gas(gas_left)
+            .raw_call(ManagedBuffer::from(SECOND_CONTRACT_REJECT_ESDT_PAYMENT))
```
Comment (andrei-marinica) [inline contracts/feature-tests/composability/esdt-contract-pair/first-contract/src/lib.rs:97]: `.raw_call(SECOND_CONTRACT_REJECT_ESDT_PAYMENT)`
also works, more compact

File: contracts/feature-tests/composability/scenarios/forw_raw_sync_echo.scen.json [hunk 2]
```diff
@@ -163,4 +158,4 @@
             }
         }
     ]
-}
+}
```
Comment (andrei-marinica) [inline contracts/feature-tests/composability/scenarios/forw_raw_sync_echo.scen.json:161]: would be nice to run `mx-scenario-go fmt .` here, to format the scenarios

File: contracts/feature-tests/rust-testing-framework-tester/src/lib.rs [hunk 3]
```diff
@@ -195,12 +196,11 @@ pub trait RustTestingFrameworkTester: dummy_module::DummyModule {
...
 
-        self.send_raw().async_call_raw(
-            &other_sc_address,
-            &BigUint::zero(),
-            &ManagedBuffer::new_from_bytes(b"add"),
-            &args,
-        );
+        self.tx()
+            .to(&other_sc_address)
+            .raw_call("call")
```
Comment (andrei-marinica) [inline contracts/feature-tests/rust-testing-framework-tester/src/lib.rs:201]: "add", not "call"

File: contracts/feature-tests/rust-testing-framework-tester/src/lib.rs [hunk 4]
```diff
@@ -195,12 +196,11 @@ pub trait RustTestingFrameworkTester: dummy_module::DummyModule {
...
+            .arguments_raw(args)
```
Comment (andrei-marinica) [inline contracts/feature-tests/rust-testing-framework-tester/src/lib.rs:202]: If it's only 1 argument, better to replace with `.argument(value)`

File: contracts/feature-tests/rust-testing-framework-tester/src/lib.rs [hunk 5]
```diff
@@ -218,13 +218,13 @@ pub trait RustTestingFrameworkTester: dummy_module::DummyModule {
...
-            &BigUint::zero(),
-            &ManagedBuffer::new_from_bytes(b"addValue"),
-            &args,
-        );
+        let gas_left = self.blockchain().get_gas_left();
+        self.tx()
+            .to(&other_sc_address)
+            .gas(gas_left)
+            .raw_call("addValue")
+            .arguments_raw(args)
```
Comment (andrei-marinica) [inline contracts/feature-tests/rust-testing-framework-tester/src/lib.rs:226]: If it's only 1 argument, better to replace with .argument(value)

File: framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs [hunk 6]
```diff
@@ -79,6 +79,41 @@ where
     }
 }
 
+impl<Api, Payment, Gas, FromSourceValue, RH>
+    Tx<
+        TxScEnv<Api>,
+        (),
+        &ManagedAddress<Api>,
```
Comment (andrei-marinica) [inline framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs:86]: Not like this. To generalize, make `To` generic, make it implement `TxToSpecified`, which has method `with_value_ref`

File: framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs [hunk 7]
```diff
@@ -133,3 +168,58 @@ where
         });
     }
 }
+
+impl<Api, Payment, Gas, RH>
```
Comment (andrei-marinica) [inline framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs:172]: Same here. Generalize using `TxToSpecified`, not by duplicating the entire call.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by BiancaIalangi — APPROVED** [review]

**Pull request merged**